File: version/1.18.2/src/main/java/com/github/sorusclient/client/adapter/v1_18_2/SessionImpl.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_18_2;

import com.github.sorusclient.client.adapter.ISession;
import v1_18_2.net.minecraft.client.MinecraftClient;

public class SessionImpl implements ISession {

    @Override
    public String getUUID() {
        return MinecraftClient.getInstance().getSession().getUuid();
    }

    @Override
    public String getAccessToken() {
        return MinecraftClient.getInstance().getSession().getAccessToken();
    }

    @Override
    public String getUsername() {
        return MinecraftClient.getInstance().getSession().getUsername();
    }

}

File: version/1.18.2/src/main/kotlin/com/github/sorusclient/client/adapter/v1_18_2/SessionImpl.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_18_2

import com.github.sorusclient.client.adapter.ISession
import v1_18_2.net.minecraft.client.MinecraftClient

class SessionImpl: ISession {

    override fun getUUID(): String {
        return MinecraftClient.getInstance().session.uuid
    }

    override fun getAccessToken(): String {
        return MinecraftClient.getInstance().session.accessToken
    }

    override fun getUsername(): String {
        return MinecraftClient.getInstance().session.username
    }

}